cmd/env/push/push.go

```diff
@@ -182,19 +188,33 @@ func newService(envService env.EnvServicer, db database.Database, vinzService vi
 	}
 }
 
-func (s *service) putEnv(orgID, envName, appID string, e models.Env, secret models.Secret, config config.Config, secretKeyId int64, skippedEnvs *[]string) (err error, skippable bool) {
+func (s *service) putEnv(orgID, envName, appID string, e models.Env, currentSecret models.Secret, config config.Config, skippedEnvs *[]string) (err error, skippable bool) {
 	// Check local environment
 	currentLocalEnv, exists := s.db.GetSecret(database.SecretKey{
 		ProjectId: *config.ProjectId,
 		AppId:     *config.AppId,
 		EnvName:   envName,
 	})
+
+	var replacingSecretKeyID int64 = 0
 	if exists {
-		err, skippable := s.validateLocalEnv(&currentLocalEnv, &e, secret)
+		err, skippable := s.validateLocalEnv(&currentLocalEnv, &e, currentSecret)
 		if err != nil {
 			return err, false
 		}
-		if skippable && secret.SecretKeyId == secretKeyId {
+
+		// get the latest secret key id for this env, if it exists
+		latestEnv, err := s.envService.GetEnvironmentEnv(orgID, appID, envName, nil, nil)
```

Comment:
Instead of trusting the secret key ID we have been passing through is right (spoiler, it never way before with rotate), fetch the latest ENV for this and use that secret key ID. That's what we're trying to replace regardless.